pyodide-build/pyodide_build/testing.py

```diff
@@ -1,29 +1,79 @@
+import ast
 import contextlib
-import inspect
-from base64 import b64encode
-from typing import Callable, Collection
+import pickle
+import sys
+import traceback
+from base64 import b64decode, b64encode
+from typing import Any, Callable, Collection
 
 import pytest
 
 
-def _run_in_pyodide_get_source(f):
-    lines, start_line = inspect.getsourcelines(f)
-    num_decorator_lines = 0
-    for line in lines:
-        if line.startswith("def") or line.startswith("async def"):
-            break
-        num_decorator_lines += 1
-    start_line += num_decorator_lines - 1
-    # Remove first line, which is the decorator. Then pad with empty lines to fix line number.
-    lines = ["\n"] * start_line + lines[num_decorator_lines:]
-    return "".join(lines)
-
-
-def chunkstring(string, length):
-    return (string[0 + i : length + i] for i in range(0, len(string), length))
-
-
-from pprint import pformat
+def _encode_ast(module_ast, funcname):
+
+    nodes: list[Any] = []
+    for node in module_ast.body:
+        # We need to include the magic imports that pytest inserts
+        if (
+            isinstance(node, ast.Import)
+            and node.names[0].asname
+            and node.names[0].asname.startswith("@")
```

Comment:
Actual imports starting with an @. The @ is included to ensure that they aren't valid identifiers so that they are difficult to use without an ast transform. This reduces the chance they will clash with the identifiers used in the module.

They are injected into the ast here:
https://github.com/pytest-dev/pytest/blob/5f9d68c8d93b792116fc474cb4ec3072b7baa6f1/src/_pytest/assertion/rewrite.py#L716